DFE-Digital/curriculum-materials
# Updating seeds to seed resources

Review after https://github.com/DFE-Digital/curriculum-materials/pull/136 has been merged
Feel free to merge this if it's approved as other PRs will be built off it

### Context
As part of the work for adding previews for open document files we need to update the seeder to write to the new models.

### Changes proposed in this pull request
Updates the api to write to the new resource models.
Updates the seeder to write to the new resource models.
Adds previews to the existing content.

When run in API mode the seeder will only write to the new models.
When run in non API mode the seeder will write to both the old and new models, so use non API mode until the rest of the changes are in.

The seeds now contain previews of the resources in the format resource-`name.preview.ext`

```
│       │   │   ├── prelude-to-the-battle
│       │   │   │   ├── 1
│       │   │   │   │   ├── keyword-matching
│       │   │   │   │   │   ├── pupil
│       │   │   │   │   │   │   ├── normans.odt
│       │   │   │   │   │   │   ├── normans.preview.pdf
│       │   │   │   │   │   │   ├── the-house-of-normandy.pdf
│       │   │   │   │   │   │   └── the-house-of-normandy.preview.pdf
│       │   │   │   │   │   ├── slides.odp
│       │   │   │   │   │   ├── slides.preview.png
│       │   │   │   │   │   └── teacher
│       │   │   │   │   │       ├── norman-conquest.gif
│       │   │   │   │   │       └── norman-conquest.preview.gif
```

A preview file is not required for a resource. Even for resources that the browser can display inline, such as pdfs, I've added a preview copy. This convention saves code complexity in the seeder at the cost of unnecessary duplication of a couple of files. Space on S3 is cheap and we can always revisit this later.

### Guidance to review
Manual testing:
Flatten the database and run the seeds in api mode, write a preview to disk in the rails console `File.binwrite 'test', TeacherResource.first.preview.download` it should render correctly.
Flatten the database and run seeds in non api mode, write a preview to disk in the rails console `File.binwrite 'test', TeacherResource.first.preview.download` it should render correctly and the rest of the app should still function as before.
